src/assetloading/asset_loader.rs

```diff
@@ -11,65 +11,64 @@ pub struct AssetManager<T> {
 }
 
 impl<T> Default for AssetManager<T>
-where T:Asset
+where
+    T: Asset,
 {
     fn default() -> Self {
         Self::new()
     }
 }
 
 impl<T> AssetManager<T>
-where T:Asset
+where
+    T: Asset,
 {
     pub fn new() -> Self {
         AssetManager {
             assets: HashMap::new(),
         }
     }
 
-    pub fn get_asset_handle(&self, path: &str) -> Option<Handle<T>>
-    {
+    pub fn get_asset_handle(&self, path: &str) -> Option<Handle<T>> {
         self.assets.get(path).cloned()
     }
 
-    pub fn get_asset<'a>(&self, path: &str, storage: &'a AssetStorage<T>) -> Option<&'a T>
-        where
-            T: Asset,
-    {
-        if let Some(h) = self.get_asset_handle(path) {
-            storage.get(&h)
-        } else {
-            None
-        }
-    }
-
-    pub fn get_asset_handle_or_load<'a, F>(&mut self, path: &str, format: F, options: F::Options, storage: &'a mut AssetStorage<T>, loader: &Loader) -> Option< Handle<T>>
-        where
-            F: Format<T> + 'static,
+    pub fn get_asset_handle_or_load<'a, F>(
+        &mut self,
+        path: &str,
+        format: F,
+        options: F::Options,
+        storage: &'a mut AssetStorage<T>,
+        loader: &Loader,
+    ) -> Option<Handle<T>>
+    where
+        F: Format<T> + 'static,
     {
         if let Some(h) = self.get_asset_handle(path) {
-            return Some(h)
+            return Some(h);
         }
         if let Some(h) = self.load::<F>(path, format, options, storage, loader) {
-            return Some(h)
+            return Some(h);
         }
         None
     }
 
-    pub fn load<F>(&mut self, path: &str, format: F, options: F::Options, storage: &mut AssetStorage<T>, loader: &Loader) -> Option<Handle<T>>
-        where
-            T: Asset,
-            F: Format<T> + 'static,
+    pub fn load<F>(
+        &mut self,
+        path: &str,
+        format: F,
+        options: F::Options,
+        storage: &mut AssetStorage<T>,
+        loader: &Loader,
+    ) -> Option<Handle<T>>
```

Comment:
why `Option` if you always return `Some`?

Reply:
will change as soon, as the amethyst loader has proper errorhandling with options too